include/irr/asset/CTexturePacker.h

```diff
@@ -163,83 +133,144 @@ class ITexturePacker
 
 class ICPUTexturePacker : protected ITexturePacker
 {
-    uint32_t megaimgLayerSize(uint32_t _pgSz, uint32_t _tilesPerDim) const
+    uint32_t physAddrTexLayerSz(uint32_t _pgSz, uint32_t _tilesPerDim) const
     {
         return _tilesPerDim*(TILE_PADDING + _pgSz) + TILE_PADDING;
     }
-    core::vector2du32_SIMD pageCoords(uint32_t _x, uint32_t _y, uint32_t _pgSz) const
+    core::vector3du32_SIMD pageCoords(STexOffset _txoffset, uint32_t _pgSz) const
     {
-        core::vector2du32_SIMD coords(_x,_y);
+        core::vector3du32_SIMD coords(_txoffset.x(),_txoffset.y(), 0u);
         coords *= (_pgSz + TILE_PADDING);
         coords += TILE_PADDING;
+        coords.z = _txoffset.layer();
         return coords;
     }
```

Comment:
wrong the offsets into the padded physical texture are stored at the pixels of the page table

Reply:
what does it have to do with this code?

Reply:
the fact that you were using `STexOffset` as the input to this function and I'm under the impression that `STexOffset` is what you get back from the `alloc(ICPUImage,Subresource)` function when you ask the Packer to place an image and its mipmap chain in the virtual address space

Reply:
ah, so it's just the confusion from me using STexOffset for both things

Reply:
looks like it